go/vt/mysqlctl/schema.go

```diff
@@ -536,6 +536,10 @@ func (mysqld *Mysqld) ApplySchemaChange(ctx context.Context, dbName string, chan
 		sql = "SET sql_log_bin = 0;\n" + sql
 	}
 
+	if change.SetForeignKeyChecksOff {
+		sql = "SET foreign_key_checks = 0;\n" + sql
```

Comment:
This variable has a `SESSION` and `GLOBAL` context and value. We should be explicit here and use session:
```
SET @@session.foreign_key_checks = 0;\n" + sql
```

https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_foreign_key_checks

Reply:
The syntax, as it is, _is_ setting the session value only. In MySQL, if neither `session` nor `global` is specified, then it is presumed to be `session`. For example:


```sql
> set read_only=1;
ERROR 1229 (HY000): Variable 'read_only' is a GLOBAL variable and should be set with SET GLOBAL
```

So I think it's fine to leave as it is.

Reply:
Yes, it’s not about the actual effect in 8.0 but rather being explicit about what your intention is.

Reply:
What my thinking is, that as MySQL veterans this is explicit enough. It's common practice to say `set foreign_key_checks=0` and it's clear that you're intent on modifying the session variable.

Reply:
I have the same opinion/reasoning as Shlomi